packages/datetime/src/dateInput.tsx

```diff
@@ -327,34 +327,38 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
      * This is a no-op unless moment-timezone's setDefault has been called.
      */
     private fromMomentToDate = (momentDate: moment.Moment) => {
-        return new Date(
-            momentDate.year(),
-            momentDate.month(),
-            momentDate.date(),
-            momentDate.hours(),
-            momentDate.minutes(),
-            momentDate.seconds(),
-            momentDate.milliseconds(),
-        );
+        if (momentDate == null) {
+            return undefined;
```

Comment:
nit: Do we care about returning `undefined` for an `undefined` param and `null` for a `null` param? That's the pattern I tend to prefer. Not a big deal to me though.